packages/expo-location/android/src/main/java/expo/modules/location/LocationModule.java

```diff
@@ -463,7 +463,12 @@ public void hasServicesEnabledAsync(final Promise promise) {
 
   @ExpoMethod
   public void startLocationUpdatesAsync(String taskName, Map<String, Object> options, final Promise promise) {
-    if (isMissingBackgroundPermissions()) {
+    boolean useForegroundService = LocationTaskConsumer.useForegroundService(options);
+
+    // There are two ways of starting this service.
+    // 1. As a background location service, this requires the background location permission.
+    // 2. As a user-initiated foreground service with notification, this does NOT require the background location permission.
+    if (!useForegroundService && isMissingBackgroundPermissions()) {
```

Comment:
Renaming to `shouldUseForegroundService`:

```suggestion
    if (!shouldUseForegroundService && isMissingBackgroundPermissions()) {
```